### PR TITLE
I1 visitor nav

### LINKED ISSUE #2: User Story 2, Visitor Navigation
As a visitor
I see a navigation bar
This navigation bar includes links for the following:
- a link to return to the welcome / home page of the application ("/")
- a link to browse all items for sale ("/items")
- a link to see all merchants ("/merchants")
- a link to my shopping cart ("/cart")
- a link to log in ("/login")
- a link to the user registration page ("/register")

Next to the shopping cart link I see a count of the items in my cart

### PR DESCRIPTION
**Pull request by ryandbarnett**
**Spec**
Add nav link tests
Update user new spec to user register path instead of users/new

**Routes**
Add home, login routes
Change /users/new get and post to /register

**Controllers**
Add sessions controller w new action
Add index action to application controller
Update users controller to use render show instead of redirect to

**Views**
Add login, home, and register links to nav bar
Add blank home page view
Add blank login page view
Update form for to be form with
Update form url to be register instead of users/new
Delete center html element

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -35,7 +35,9 @@
   post "/orders", to: "orders#create"
   get "/orders/:id", to: "orders#show"
 
-  get "/users/new", to: "users#new"
-  post "/users/new", to: "users#create"
+  get "login", to: "sessions#new"
+
+  get "/register", to: "users#new"
+  post "/register", to: "users#create"
```
Comment (PhilipDeFraties) [inline config/routes.rb:42]: The only problem I see with this is that by changing the path to '/register' we lose the ability to refactor this using `resources: :users`

File: config/routes.rb [hunk 2]
```diff
@@ -35,7 +35,9 @@
   post "/orders", to: "orders#create"
   get "/orders/:id", to: "orders#show"
 
-  get "/users/new", to: "users#new"
-  post "/users/new", to: "users#create"
+  get "login", to: "sessions#new"
```
Comment (PhilipDeFraties) [inline config/routes.rb]: missing the /

**Review by PhilipDeFraties — APPROVED** [review]
Its pretty confusing that they specify a specific route path for new user creation as being '/register' yet they want us to refactor using resources, dont really understand that. Looks good though overall.

File: spec/features/navigation_spec.rb [hunk 3]
```diff
@@ -1,46 +1,66 @@
...
+  end
+
+  it 'I see a register link' do
+    visit '/'
 
-      within 'nav' do
-        expect(page).to have_content("Cart: 0")
-      end
+    within 'nav' do
+      click_link 'Register'
```
Comment (ryandbarnett) [inline spec/features/navigation_spec.rb:43]: There should be an expect page to have link register !

**Review by D-Lessenden — APPROVED** [review]

**Pull request merged**